### DISCUSSION TITLE
I18n language flow

### DISCUSSION THREAD

**Discussion by sakulstra** [aave/interface#576] [General]
Huge parts of the aave community are non-native english speakers so It would obviously be nice to provide the ui in their native language. While localization is nice I think there are some technical/maintenance things to consider here. [Here's the current incomplete flow](https://github.com/aave/interface/blob/main/CONTRIBUTING.md)

Translations have to be:
- in good **quality** as bad translations do more harm than help. Also they have to be proofread to prevent abusing them for inappropriate messaging.
- have a big enough **user-base** as adding a language adds continuous maintenance burden (also after initial addition)
- have to be **complete** so that you don't have a wild en-xx mixture on the screen

To fulfill these goals there should be a defined flow for listing & unlisting languages on common and well understood criteria.

**Comment by sakulstra** [comment]
My suggestion to address these topics:
1. we should have at least two translators per language that we want to support. Translations should be added via crowdin so we can have a proper approval & collaboration workflow per language.

2.  ~~this is a rather complicated point imo. Some suggestions which all have limitations:
2.1 Do a governance snapshot vote per language. This has the drawback of excluding people from the vote that don't have aave tokens (which is a substantial part of the aave user base). Also token weighted voting doesn't seem fair in that context. 
2.2 Do a :+1: sentiment poll on github & define a threshold needed to consider the addition. Consider = allow translation if the other points are fulfilled. The drawback is that non github users would be excluded from the vote (creating a github account is free, so perhaps not so bad). This would be my personal preference to have it in one place.
2.3 Do a :heart: sentiment poll  on https://governance.aave.com/c/other/6 . The drawback is that non forum users are excluded from the vote (creating a forum account is free, so perhaps not so bad).~~
3. I'd recommend a threshold of 80% - when 80% is translated it should be enabled on the ui. If % drops it should be disabled again.

**Reply by Xk9eboF6** [reply]
> My suggestion to address these topics:
> 
> 1. we should have at least two translators per language that we want to support. Translations should be added via crowding so we can have a proper approval & collaboration workflow per language.
> 2. this is a rather complicated point imo. Some suggestions which all have limitations:
>    2.1 Do a governance snapshot vote per language. This has the drawback of excluding people from the vote that don't have aave tokens (which is a substantial part of the aave user base). Also token weighted voting doesn't seem fair in that context.
>    2.2 Do a 👍 sentiment poll on github & define a threshold needed to consider the addition. Consider = allow translation if the other points are fulfilled. The drawback is that non github users would be excluded from the vote (creating a github account is free, so perhaps not so bad). This would be my personal preference to have it in one place.
>    2.3 Do a ❤️ sentiment poll  on https://governance.aave.com/c/other/6 . The drawback is that non forum users are excluded from the vote (creating a forum account is free, so perhaps not so bad).
> 3. I'd recommend a threshold of 80% - when 80% is translated it should be enabled on the ui. If % drops it should be disabled again.

Completely agree with point 1, point 2.3 and point 3.

**Reply by sakulstra** [reply]
@Xk9eboF6 would be up for doing a pilot with this approach to see if it can work out?
Creating a governance post with a sentiment poll & asking for other translators.
If there are 2-3 translators including you I think @foodaka could hook you up with crowdin translator accounts.

---

Thinking about this a bit more, I have no idea how much "support" we should expect on the sentiment part though as we have 0 data from before. Also people active in the forum usually have good understanding skills of english anyways so they might not care for a translated ui at all. Because of that i wouldn't set any threshold beforehand & just collect some data in this pilot.

**Reply by Xk9eboF6** [reply]
> @Xk9eboF6 would be up for doing a pilot with this approach to see if it can work out? Creating a governance post with a sentiment poll & asking for other translators. If there are 2-3 translators including you I think @foodaka could hook you up with crowdin translator accounts.
> 
> Thinking about this a bit more, I have no idea how much "support" we should expect on the sentiment part though as we have 0 data from before. Also people active in the forum usually have good understanding skills of english anyways so they might not care for a translated ui at all. Because of that i wouldn't set any threshold beforehand & just collect some data in this pilot.

The thing Is that we got a grant so we have to do this work :)

**Reply by sakulstra** [reply]
Oh that's unfortunate 😅
My i ask for how big the grant is?
Does it require a certain amount of time of maintenance or is it just initial translation - which might start missing stuff tomorrow?

**Reply by Xk9eboF6** [reply]
> Oh that's unfortunate 😅
> My i ask for how big the grant is?
> Does it require a certain amount of time of maintenance or is it just initial translation - which might start missing stuff tomorrow?

The grant is both for the app and a community we have setup at https://www.aave.gr
I don't know if  I can disclose the amount ($).

Well, we'll keep it up to date. And if big changes arise we may ask for a another grant.

**Reply by ctrlaltf** [reply]
> My suggestion to address these topics:
> 
> 1. we should have at least two translators per language that we want to support. Translations should be added via crowdin so we can have a proper approval & collaboration workflow per language.
> 2. this is a rather complicated point imo. Some suggestions which all have limitations:
>    2.1 Do a governance snapshot vote per language. This has the drawback of excluding people from the vote that don't have aave tokens (which is a substantial part of the aave user base). Also token weighted voting doesn't seem fair in that context.
>    2.2 Do a 👍 sentiment poll on github & define a threshold needed to consider the addition. Consider = allow translation if the other points are fulfilled. The drawback is that non github users would be excluded from the vote (creating a github account is free, so perhaps not so bad). This would be my personal preference to have it in one place.
>    2.3 Do a ❤️ sentiment poll  on https://governance.aave.com/c/other/6 . The drawback is that non forum users are excluded from the vote (creating a forum account is free, so perhaps not so bad).
> 3. I'd recommend a threshold of 80% - when 80% is translated it should be enabled on the ui. If % drops it should be disabled again.

Agreed with point 1 and 3. As for the points in 2, to be discussed, as i see a lot of logistics complications with point 2.1, which would elongate the process before it even gets started, similar to 2.3 and the exclusion of others to vote.

2.2 may be viable as sentiment on which language should be next. the people would need to be vetted one way or another before being added to crowdin and how do we make them accountable to translate properly? We would need a third person perhaps to verify the translations are done correctly. 

I do not see the translations to be a relevant request for grants also.

**Reply by sakulstra** [reply]
Agreed, perhaps it makes sense to just strike 2 all together.

If someone in the community thinks it's reasonable to translate to `x` it doesn't really make sense to put roadblocks in the way. When the community has interest in a certain language they will maintain it - if not, threshold at some point drops below 80% and it gets removed. Perhaps that's all the control needed.

**Reply by sendra** [reply]
So how should access to crowdin be asked then? Does it make sense to just create an issue with Translation <Language> in the title, and then explain a bit why translation is needed, and a bit of the Translator background?

This issue then could be linked with a pr to activate the language + flag svg

**Comment by foodaka** [comment]
I agree with most topics suggested above from @sakulstra

1) Having 2 translators per language will ensure a higher quality of translations as well as keep out spam.
2) On the snapshot it might be a bit too much of an overhead to just add a language. It might be easiest to do it through a github poll. Open to more ideas or suggestions on 2.
3) We should not include a language that does not have at least 80% of the strings translated

There can be some incentives from the aave grants dao to make sure we get quality translations.

**Comment by Xk9eboF6** [comment]
> Huge parts of the aave community are non-native english speakers so It would obviously be nice to provide the ui in their native language. While localization is nice I think there are some technical/maintenance things to consider here. [Here's the current incomplete flow](https://github.com/aave/interface/blob/main/CONTRIBUTING.md)
> 
> Translations have to be:
> 
> * in good **quality** as bad translations do more harm than help. Also they have to be proofread to prevent abusing them for inappropriate messaging.
> * have a big enough **user-base** as adding a language adds continuous maintenance burden
> * have to be **complete** so that you don't have a wild en-xx mixture on the screen
> 
> To fulfill these goals there should be a defined flow for listing & unlisting languages on common and well understood criteria.

Completely agree.

**Reply by Pasucaru** [reply]
Agree , I am french speaker and I alreday done some translation from english to french, I tested the french version of the new testnet gho.aave.com and I saw some mistake
